src/nnet2/nnet-component.cc

```diff
@@ -4110,6 +4124,8 @@ void Convolutional1dComponent::Read(std::istream &is, bool binary) {
   ReadBasicType(is, binary, &patch_step_);
   ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<PatchStride>");
   ReadBasicType(is, binary, &patch_stride_);
+  ExpectOneOrTwoTokens(is, binary, ostr_beg.str(), "<RearrangeInput>");
```

Comment:
I would recommend using variable names which do not require the user to read the code to understand their usage. I would recommend using a variable which specifies in detail what is the default vectorization order and when users would have to rearrange_input.

I tried to solve this issue by using input_vectorization variable in ConvolutionComponent being written for nnet3. (see #283, @[line](https://github.com/vijayaditya/kaldi/blob/05b00b8834d3c45baed4461dc923109da994c3cb/src/nnet3/nnet-simple-component.h#L922) ) The documentation for this component is not complete, but I basically tried to explain that zyx is the default vectorization order and users can specify that they are using yzx input. Based on this info the component decides if it has to rearrange the input or not.